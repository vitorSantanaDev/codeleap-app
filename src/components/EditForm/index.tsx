import React, { FormEvent, useState } from 'react'

import Button from 'components/Button'
import Heading from 'components/Heading'
import TextInput from 'components/TextInput'
import TextAreaInput from 'components/TextAreaInput'

import * as BaseFormLayout from 'components/Layouts/BaseFormLayout'

const EditForm: React.FC = () => {
  const [inputsValues, setInputValues] = useState({ title: '', content: '' })

  const handleInputChange = (fieldName: string, value: string) => {
    setInputValues((prevState) => ({ ...prevState, [fieldName]: value }))
  }

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault()
    console.log({ inputsValues })
  }

  return (
    <BaseFormLayout.FormWrapper>
      <Heading>Edit item</Heading>
      <BaseFormLayout.Form onSubmit={handleSubmit}>
        <TextInput
          name="title"
          label="Title"
          placeholder="Title"
          handleChange={(value) => handleInputChange('title', value)}
        />
        <TextAreaInput
          name="content"
          label="Content"
          placeholder="Content"
          handleChange={(value) => handleInputChange('content', value)}
        />
        <BaseFormLayout.ButtonWrapper>
          <Button
            onClick={() => console.log('Cancel edition')}
            type="submit"
            bgColor="white"
          >
            Cancel
          </Button>

          <Button type="submit" bgColor="green">
            Save
          </Button>
        </BaseFormLayout.ButtonWrapper>
      </BaseFormLayout.Form>
    </BaseFormLayout.FormWrapper>
  )
}

export default EditForm
