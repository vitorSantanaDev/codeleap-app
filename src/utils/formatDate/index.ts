import {
  format,
  parseISO,
  minutesToHours,
  secondsToMinutes,
  differenceInSeconds
} from 'date-fns'

export const formattingStringWithTheTimePassedBetweenTwoDate = (
  hours: number,
  minutes: number,
  seconds: number
): string => {
  if (hours < 1 && minutes < 1) {
    return `${seconds} seconds ago`
  }

  let formattedTime = ''

  if (hours > 0) {
    formattedTime = `${hours} ${hours > 1 ? 'hours' : 'hour'} ago`
  } else if (hours < 1) {
    formattedTime = `${minutes} minutes ago`
  }

  return formattedTime
}

export const rangeOfTimePassedBetweenTwoDates = (
  start: string,
  end: string
): string => {
  const startDate = parseISO(start)
  const endDate = parseISO(end)

  const allSeconds = differenceInSeconds(endDate, startDate)
  const allMinutes = secondsToMinutes(allSeconds)

  const minutesInSeconds = allMinutes * 60
  const seconds = allSeconds - minutesInSeconds

  const hours = minutesToHours(allMinutes)

  const hoursInMinutes = hours * 60

  const minutes = allMinutes - hoursInMinutes

  return hours > 24
    ? format(new Date(startDate), 'dd/MM/yyyy')
    : formattingStringWithTheTimePassedBetweenTwoDate(hours, minutes, seconds)
}
