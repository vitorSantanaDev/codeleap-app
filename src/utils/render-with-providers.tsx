import { Provider } from 'react-redux'
import React, { PropsWithChildren } from 'react'

import theme from 'styles/theme'
import { ThemeProvider } from 'styled-components'
import { RenderOptions, render } from '@testing-library/react'

import { RootState } from 'redux/store'
import { userReducer } from 'redux/actions/user-slice'
import { PreloadedState, configureStore } from '@reduxjs/toolkit'
import { QueryClient, QueryClientProvider } from 'react-query'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialState?: PreloadedState<RootState>
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  store?: any
}

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    initialState = {},
    store = configureStore({
      reducer: { user: userReducer },
      preloadedState: initialState
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <QueryClientProvider client={new QueryClient()}>
        <Provider store={store}>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </Provider>
      </QueryClientProvider>
    )
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}
