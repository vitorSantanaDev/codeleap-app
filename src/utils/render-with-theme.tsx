import React, { ReactElement } from 'react'

import theme from 'styles/theme'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

export const renderWithTheme = (component: ReactElement) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
