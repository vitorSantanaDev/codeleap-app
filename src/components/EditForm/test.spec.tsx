import React from 'react'
import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/render-with-theme'

import EditForm from '.'

describe('<EditForm />', () => {
  it('should render the form', () => {
    renderWithTheme(<EditForm />)

    const titleInput = screen.getByPlaceholderText(/title/i)
    expect(titleInput).toBeInTheDocument()

    const contentInput = screen.getByPlaceholderText(/content/i)
    expect(contentInput).toBeInTheDocument()

    const saveButton = screen.getByRole('button', { name: /save/i })
    expect(saveButton).toBeInTheDocument()

    const cancelButton = screen.getByRole('button', { name: /cancel/i })
    expect(cancelButton).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<EditForm />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
