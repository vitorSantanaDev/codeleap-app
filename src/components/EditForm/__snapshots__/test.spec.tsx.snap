// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditForm /> should match snapshot 1`] = `
.c6 {
  cursor: pointer;
  padding: 0.6rem 2.7rem;
  font-weight: 700;
  font-size: 1.6rem;
  border-radius: 0.8rem;
  background-color: #FFFFFF;
  color: #000000;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  border: 1px solid #999999;
}

.c6:hover {
  color: #fff;
}

.c6:disabled {
  cursor: not-allowed;
  -webkit-filter: saturate(30%);
  filter: saturate(30%);
}

.c6:hover {
  color: #808080;
}

.c7 {
  cursor: pointer;
  padding: 0.6rem 2.7rem;
  font-weight: 700;
  font-size: 1.6rem;
  border-radius: 0.8rem;
  background-color: #47B960;
  color: #FFFFFF;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  border: none;
}

.c7:hover {
  color: #6cc780;
}

.c7:disabled {
  cursor: not-allowed;
  -webkit-filter: saturate(30%);
  filter: saturate(30%);
}

.c7:hover {
  color: #6cc780;
}

.c0 {
  color: #000000;
  font-weight: 700;
  font-size: calc(1.6rem + 0.2rem);
}

.c2 {
  width: 100%;
  display: block;
  outline: none;
  color: #000000;
  padding: 0.8rem;
  border-radius: 0.8rem;
  border: 1px solid #777777;
}

.c2::-webkit-input-placeholder {
  color: #CCCCCC;
}

.c2::-moz-placeholder {
  color: #CCCCCC;
}

.c2:-ms-input-placeholder {
  color: #CCCCCC;
}

.c2::placeholder {
  color: #CCCCCC;
}

.c1 {
  display: inline-block;
  color: #000000;
  font-size: 1.6rem;
  margin-bottom: 0.8rem;
}

.c4 {
  width: 100%;
  resize: none;
  outline: none;
  height: 7.4rem;
  display: block;
  color: #000000;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 0.8rem;
  border-radius: 0.8rem;
  border: 1px solid #777777;
}

.c4::-webkit-input-placeholder {
  color: #CCCCCC;
}

.c4::-moz-placeholder {
  color: #CCCCCC;
}

.c4:-ms-input-placeholder {
  color: #CCCCCC;
}

.c4::placeholder {
  color: #CCCCCC;
}

.c3 {
  display: inline-block;
  color: #000000;
  font-size: 1.6rem;
  margin-bottom: 0.8rem;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  gap: 0.8rem;
}

<div
  class=""
>
  <h2
    class="c0"
    color="black"
  >
    Edit item
  </h2>
  <form
    class=""
  >
    <div
      class=""
    >
      <label
        class="c1"
        for="title"
      >
        Title
      </label>
      <input
        class="c2"
        id="title"
        name="title"
        placeholder="Title"
        type="text"
        value=""
      />
    </div>
    <div
      class=""
    >
      <label
        class="c3"
        for="content"
      >
        Content
      </label>
      <textarea
        class="c4"
        id="content"
        name="content"
        placeholder="Content"
      />
    </div>
    <div
      class="c5"
    >
      <button
        class="c6"
        type="submit"
      >
        Cancel
      </button>
      <button
        class="c7"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;
